ui/src/app/jobs/jobs.service.ts

```diff
@@ -1,39 +1,83 @@
 import { Injectable } from '@angular/core';
-import { Http, Response } from '@angular/http';
+import { Http, Response, RequestOptionsArgs } from '@angular/http';
+
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 
+import { ErrorHandler, Page } from '../shared/model';
+import { PageRequest } from '../shared/model/pagination/page-request.model';
+
+import { JobExecution } from './model/job-execution.model';
+import { HttpUtils } from '../shared/support/http.utils';
+
+import * as moment from 'moment';
+import { Moment } from 'moment';
+
 @Injectable()
 export class JobsService {
 
-  private appstUrl = '/apps';
+  private jobExecutionsUrl = '/jobs/executions';
+  public jobExecutions: Page<JobExecution>;
+  public remotelyLoaded = false;
 
-  constructor(private http: Http) { }
+  constructor(private http: Http, private errorHandler: ErrorHandler) { }
 
-  getApps(): Observable<any[]> {
-    return this.http.get(this.appstUrl)
-                    .map(this.extractData)
-                    .catch(this.handleError);
-  }
+  getJobExecutions(reload?: boolean): Observable<Page<JobExecution>> {
+    console.log(`Get Job Executions - reload ${reload}`, this.jobExecutions);
+    if (!this.jobExecutions || reload) {
+      if (!this.jobExecutions) {
+        this.jobExecutions = new Page<JobExecution>();
+      }
+      console.log('Fetching Job Executions remotely.');
+      this.remotelyLoaded = true;
 
-  private extractData(res: Response) {
-    const body = res.json();
-    return body._embedded;
-    // return body.data || { };
-  }
+      const params = HttpUtils.getPaginationParams(this.jobExecutions.pageNumber, this.jobExecutions.pageSize);
+      const requestOptionsArgs: RequestOptionsArgs = {};
+
+      return this.http.get(this.jobExecutionsUrl, { search: params })
+                    .map(this.extractData.bind(this))
+                    .catch(this.errorHandler.handleError);
 
-  private handleError (error: Response | any) {
-    // In a real world app, you might use a remote logging infrastructure
-    let errMsg: string;
-    if (error instanceof Response) {
-      const body = error.json() || '';
-      const err = body.error || JSON.stringify(body);
-      errMsg = `${error.status} - ${error.statusText || ''} ${err}`;
     } else {
-      errMsg = error.message ? error.message : error.toString();
+      this.remotelyLoaded = false;
+      console.log('Fetching Job Executions from local state.', this.jobExecutions);
```

Comment:
With Job Executions we want to see the latest state upon tab selection, not the saved state.  Because the job execution state could have changed since the last time they were on the tab.